### ISSUE TITLE
grunt-ts can't detect --noImplicitAny error with TypeScript master/HEAD

### ISSUE THREAD

**Issue by vvakame** [TypeStrong/grunt-ts#182]
```
$ node typescript/bin/tsc.js --noImplicitAny foo.ts
foo.ts(14,1): error TS7009: 'new' expression, whose target lacks a construct signature, implicitly has an 'any' type.
$ echo $?
2
```

grunt-ts only checking exit code 1.
https://github.com/TypeStrong/grunt-ts/blob/master/tasks-internal/ts.ts#L243

FYI
https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#diagnostic-message-codes
https://github.com/Microsoft/TypeScript/blob/9f0e85cb8e8409a9fbd89e0f00e45c7d61750c49/src/compiler/tsc.ts#L220-L255
https://github.com/Microsoft/TypeScript/blob/9f0e85cb8e8409a9fbd89e0f00e45c7d61750c49/src/compiler/types.ts#L815

**Comment by bparadie** [comment]
Yes, `result.code !== 0` does fix this problem.
Please fix it, it's annoying!

Run this for locally patching `ts.ts`:
```
sed -i.bak  's/result.code === 1/result.code !== 0/' node_modules/grunt-ts/tasks/ts.ts
```

**Comment by nycdotnet** [comment]
We fixed this in PR #190 but I guess I didn't update the tasks-internal?  @basarat is that required before push to NPM?

**Comment by bparadie** [comment]
@nycdotnet Not sure how your deployment chain works. I just did a fresh `npm install` and that still had the bug. Also, this issue hasn't been closed  yet, which is probably also a sign that your fix hasn't landed yet.

But thanks for fixing this issue!

**Comment by basarat** [comment]
> but I guess I didn't update the tasks-internal

Not required. Tasks-internal is *only* used in *our* compile i.e. compile `grunt-ts` TypeScript code to JavaScript.

**Comment by nycdotnet** [comment]
@vvakame and @bparadie Thanks for reporting this.  I am sorry that it took me so long to reply.  I believe that this is actually fixed correctly in PR #190.  I didn't update "tasks-internal" as part of that PR, but it is not strictly necessary.  If you are still having a problem while using v2.0.1, I think you may be seeing that grunt-ts will (by default) halt the Grunt build pipeline in the presence of type errors.  There is a grunt-ts configuration option called `failOnTypeErrors` (note that it is plural) that if set to `false` will not fail the build in the presence of mere type errors.  If I am not mistaken, that option was either not documented, not working, or both up until recently.

https://github.com/TypeStrong/grunt-ts#failontypeerrors

If you are still having a problem after looking at failOnTypeErrors, please reopen this issue and let me know.  Thanks!
